app/services/castle_risk_guard.rb

```diff
@@ -0,0 +1,18 @@
+class CastleRiskGuard < Clearance::SignInGuard
+  def call
+    return next_guard unless @session.castle_authentication
```

Comment:
couldn't come up with a more clever way of skipping a safeguard. It's needed unless we can guarantee that there will be an updated, lower risk score when creating a new authentication coming from the challenge or lock link